control-plane/p4RuntimeSerializer.cpp

```diff
@@ -1316,6 +1416,10 @@ class P4RuntimeEntriesConverter {
             return stringRepr(k->to<IR::Constant>(), keyWidth);
         } else if (k->is<IR::BoolLiteral>()) {
             return stringRepr(k->to<IR::BoolLiteral>(), keyWidth);
+        } else if (k->type->is<IR::Type_SerEnum>()) {
+            auto mem = k->to<IR::Member>();
+            auto name = mem->expr->toString();
+            return stringRepr(name);
```

Comment:
I doubt this is correct. You are supposed to serialize the enum member integral value, not its name.

Reply:
Please check if the information below looks good to you.

This is the serializable enum in the P4 program, viz., `testdata/p4_16_samples/v1model-digest-containing-ser-enum.p4`

```
enum bit<8> MySerEnum1 {
    foo = 28,
    bar = 17,
    gah = 42
}
``` 

The generated p4info is as follows:

```
serializable_enums {
    key: "MySerEnum1"
    value {
      underlying_type {
        bitwidth: 8
      }
      members {
        name: "foo"
        value: "28"
      }
      members {
        name: "bar"
        value: "17"
      }
      members {
        name: "gah"
        value: "42"
      }
    }
  }
```

Reply:
This part of the C++ code is the `P4RuntimeEntriesConverter`, which means it generates the table entries for immutable P4 tables. You are showing me an extract of a P4Info message. The 2 are different.

Reply:
Ah, sorry, I missed the code method you pointed me to.  I will look into changing the code and get back.

Reply:
Indeed, `testdata/p4_16_samples/table-key-serenum.p4` uses a table key which is a serializable enum and has `const entries {}`.  I checked in code to fix the `P4RuntimeEntriesConverter` for serializable enum.  I also fixed the code to generate bitwidth for serializable enum.